### PR TITLE
Issue #25140  Copy Quote/Sales Orders to new Customer/Prospect

### PR DESCRIPTION
**Pull request by andersdd**
Copy Quote document(s) to alternate customer or prospect

### CODE REVIEW

File: foundation-database/public/functions/copyquotetocustomer.sql [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    SELECT fetchQuNumber() INTO _qunumber;
+  END IF;
+
+-- Determine if supplied customer is a customer or prospect
+  SELECT * INTO _isCustomer FROM (
+    SELECT true as isCustomer from custinfo
+     WHERE cust_id = pCustomerid
+    UNION
+    SELECT false from prospect 
+     WHERE prospect_id = pCustomerid) foo;
```
Comment (gilmoskowitz) [inline foundation-database/public/functions/copyquotetocustomer.sql]:
```
_isCustomer := exists(select 1 from custinfo where cust_id=pCustomerid);
```
? It isn't clear from the code below if `_isCustomer` is even necessary. The two `select`s could be converted to a single `union` with minimal disruption as there is no overlap in ids.

File: foundation-database/public/functions/copyquotetocustomer.sql [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+  IF (NOT FOUND) THEN
+    RAISE EXCEPTION 'Invalid Customer supplied: %', pCustomerid;
+  END IF;  
+
+  IF (_isCustomer) THEN -- Customer Record
+    SELECT cust_name as name,  *
```
Comment (gilmoskowitz) [inline foundation-database/public/functions/copyquotetocustomer.sql]: Please replace `select *` with explicit column names or at least a table qualifier (e.g. `select cntct.*, addr_line1, addr_line2, ...`). See issue 24329.

File: foundation-database/public/functions/copyquotetocustomer.sql [hunk 3]
```diff
@@ -0,0 +1,163 @@
+CREATE OR REPLACE FUNCTION copyquotetocustomer(integer, integer, date)
+  RETURNS integer AS
+$BODY$
+-- Copyright (c) 1999-2014 by OpenMFG LLC  d/b/a xTuple. 
+-- See www.xtuple.com/CPAL for the full text of the software license.
+DECLARE
+  pQuheadid ALIAS FOR $1;
+  pCustomerid ALIAS FOR $2;
+  pSchedDate ALIAS FOR $3;
```
Comment (gilmoskowitz) [inline foundation-database/public/functions/copyquotetocustomer.sql]: Please switch to [named parameters](https://github.com/xtuple/qt-client/wiki/xTuple-Desktop-Client-Coding-Standards#plpgsql).

File: foundation-database/public/functions/copyquotetocustomer.sql [hunk 4]
```diff
@@ -0,0 +1,163 @@
+CREATE OR REPLACE FUNCTION copyquotetocustomer(integer, integer, date)
```
Comment (gilmoskowitz) [inline foundation-database/public/functions/copyquotetocustomer.sql]: Making this an overload of [`copyQuote`](https://github.com/xtuple/xtuple/blob/4_8_x/foundation-database/public/functions/copyquote.sql) would avoid introducing code duplication.

**Comment by shackbarth** [comment]
@andersdd merge conflict. can you upmerge from 4_9_x?

**Comment by ssshyam** [comment]
Unable to perform UAT for now: Since this PR depends on qt-client repository PR #575.

**Comment by gpazo** [comment]
@andersdd This pull needs an upmerge from 4_9_x as mentioned above, if you need help resolving conflicts contact me on slack.

**Comment by andersdd** [comment]
@gpazo Upmerge done.

File: enyo-client/application/enyo [hunk 5]
```diff
@@ -1 +1 @@
-Subproject commit 02e25652e3d81f694919f917fb9eb23105befaeb
+Subproject commit adb4dc23d096bb1648b8369e5da0ca39e43b1e0d
```
Comment (gilmoskowitz) [inline enyo-client/application/enyo]: The change to `enyo-client/application/enyo` and `lib/enyo-x/enyo` will revert Linda's Chrome 36 patch from August.

Comment (andersdd) [inline enyo-client/application/enyo]: @gilmoskowitz If I update submodules will that fix the problem?

Comment (gilmoskowitz) [inline enyo-client/application/enyo]: Probably.

Comment (andersdd) [inline enyo-client/application/enyo]: Tried a submodule sync and it does not appear to have given me anything to commit

Comment (gilmoskowitz) [inline enyo-client/application/enyo]: Bleh. Try:
```
$ cd enyo-client/application/enyo
$ git checkout 02e2565
$ cd ../../../lib/enyo-x/enyo
$ git checkout 02e2565
$ cd ../../..
$ git diff     # should show the 2 enyo's back to the proper commit
$ git commit ...
$ git push
```
I don't know why the merge would have gotten this wrong but I've seen similar results before.

**Comment by andersdd** [comment]
@gilmoskowitz  Thanks - your suggestion has worked.  New commit submitted.

**Pull request merged**